core/server/data/validation/index.js

```diff
@@ -42,6 +43,27 @@ validator.extend('isTimezone', function isTimezone(str) {
     return moment.tz.zone(str) ? true : false;
 });
 
+validator.extend('isEmptyOrWebsite', function isEmptyOrWebsite(str) {
```

Comment:
Facebook urls when you haven't chosen a vanity url use query strings for one example. Query strings are an integral part of URLs I don't think we should be banning them